### DISCUSSION TITLE
port eslint-plugin-react-compiler

### DISCUSSION THREAD

**Discussion by IWANABETHATGUY** [oxc-project/oxc#6801] [Feature Request]
https://react.dev/blog/2024/10/21/react-compiler-beta-release: Since `react-compiler` has reached beta, which means the compiler is relatively stable, maybe it is time to consider porting it.
https://github.com/facebook/react/tree/main/compiler/packages/eslint-plugin-react-compiler

**Comment by Boshen** [comment]
Public Beta is too early.

**Reply by dlehmhus** [reply]
@Boshen one year later, maybe now as RC it is stable enough? It would be awesome if the [exhaustive-deps rules](https://oxc.rs/docs/guide/usage/linter/rules/react/exhaustive-deps.html) would have a version for the comiler :)

**Comment by matthewdavi** [comment]
What's needed to make this work in oxlint? My understanding of the eslint-plugin-react-hooks, is that it uses information from the react compiler (written in typescript) to do its diagnostics. Will the upcoming typescript plugins for oxlint be able to interop more easily with the react compiler?

**Comment by taearls** [comment]
`eslint-plugin-react-compiler` appears to be deprecated now.  

The React team's [blog post announcing react compiler v1](https://react.dev/blog/2025/10/07/react-compiler-1#migrating-from-eslint-plugin-react-compiler-to-eslint-plugin-react-hooks) recommends removing `eslint-plugin-react-compiler` as a dependency and just using the new recommended rules in `eslint-plugin-react-hooks`.

**Reply by TheAlexLichter** [reply]
Good point! We track porting rules over from `eslint-plugin-react` and `eslint-plugin-react-hooks` in https://github.com/oxc-project/oxc/issues/1022

**Comment by TheAlexLichter** [comment]
Closing as the eslint plugin in question is deprecated.

We track porting rules over from `eslint-plugin-react` and `eslint-plugin-react-hooks` in #1022.
